src/libfuncs/uint128.rs

```diff
@@ -194,29 +195,33 @@ pub fn build_from_felt252<'ctx, 'this>(
     let range_check =
         super::increment_builtin_counter(context, entry, location, entry.argument(0)?.into())?;
 
-    let arg1 = entry.argument(1)?.into();
-
-    let k1 = entry.const_int(context, location, 1, 252)?;
-    let k128 = entry.const_int(context, location, 128, 252)?;
-
-    let min_wide_val = entry.append_op_result(arith::shli(k1, k128, location))?;
+    let value = entry.argument(1)?.into();
+    let bound = BigUint::from(u128::MAX) + 1u32;
+    let k_u128_max_1 = entry.const_int(context, location, bound, 252)?;
     let is_wide = entry.append_op_result(arith::cmpi(
         context,
         CmpiPredicate::Uge,
-        arg1,
-        min_wide_val,
+        value,
+        k_u128_max_1,
         location,
     ))?;
 
-    let lsb_bits = entry.append_op_result(arith::trunci(
-        arg1,
+    let value_u128 = entry.append_op_result(arith::trunci(
+        value,
         IntegerType::new(context, 128).into(),
         location,
     ))?;
 
-    let msb_bits = entry.append_op_result(arith::shrui(arg1, k128, location))?;
-    let msb_bits = entry.append_op_result(arith::trunci(
-        msb_bits,
+    let value_if_of = entry.append_op_result(arith::divui(value, k_u128_max_1, location))?;
+    let overflow = entry.append_op_result(arith::remui(value, k_u128_max_1, location))?;
```

Comment:
Isn't this the same we had earlier, but less efficient? I think the only thing that really changed is the `is_wide` comparison, which is now properly implemented (we're comparing against `u128::MAX + 1` instead of `128`).

Could you try if just shifting and truncating works now too?